COMMIT_GUIDELINES.md

```diff
@@ -0,0 +1,149 @@
+and you start using the word "and", you probably want to split the commit up into 2 or more individual commits.
+
+- **Each commit should compile** (try to make each commit self-contained so that it is possible to compile the code at
+each commit)
+
+- **Take your time** when composing the commit message. In order to have a good git-history the commit messages are essential.
+Also if you put effort into the commit-message, you'll save work when creating a PullRequest as the description is already available.
+
+
+# Commit message
+
+Commit messages must follow the scheme
+
+```
+TYPE(Scope): Summary
+
+Message Body
+
+Footer
+```
+
+The blank lines in between are mandatory. A commit **must** include a `TYPE` and a `Summary` and **may**
+additionally contain any of the other components listed here.
+
+
+## Subject line
+
+The first line ("Subject line") should not exceed 50-70 characters. This is what gets displayed on GitHub at first glance, so it
+should contain the most important information. In order for it to be as short and precise as possible, there is the `TYPE` and
+optionally a `Scope`. With these it should already be clear what this commit is about in general. The short `Summary` should
+then include further information that is important to understand the general idea of this commit at a glance.
+
+
+### TYPE
+
+The `TYPE` is one of the following:
+
+| **TYPE** | **Description** | **Example** |
+| -------- | --------------- | ----------- |
+| BREAK    | A breaking change - not backwards-compatible. | A change in protocol (e.g. change UDP message format) |
+| FEAT     | Introduction of a new feature (or extension of an existing one) | |
+| FIX      | A bug fix | |
+| FORMAT   | Change of formatting - does not influence how the code works | Change indentation of a line; Add braces around `if` body |
+| DOCS     | Changes to the documentation (either in-source or out-of-source) | Add a Doxygen comment to a function |
+| TEST     | Adds, changes or removes a test-case | |
+| MAINT    | Maintenance - Change of non-code files | Change of the README |
+| CI       | Changed something for the CI (continous integration) | Update TravisCI to use newer ubuntu version |
+| REFAC   | Code refactoring | Renamed variable `x` to `y` |
+
+The `TYPE` has to be in **all-uppercase** in order for it to stand out.
+
+If you feel like you need to use 2 or more types for a single commit but *can't split it* into multiple commits, you can
+combine types with `/`: `FEAT/CI: <Summary>`
+
+
+### Scope
+
+What area is the change about. For now we don't have fixed scope keywords. A scope could be something like `ui`, `client`,
+`server`, `ice`, `grpc`, ...
+
+
+### Summary
+
+The `Summary` is the heart of the subject line. It should contain a **very brief** summary of what you did in that commit.
+In order to make this as short as possible, you may use grammatically incorrect sentences
+("Add ability" instead of "add the ability")
```

Comment:
```suggestion
In order to make this as short as possible, you may use grammatically incorrect sentences
("Add ability" instead of "add the ability").
```

Reply:
Done